crates/rpc/src/v05/method/get_block_with_txs.rs

```diff
@@ -0,0 +1,264 @@
+    };
+
+    let storage = context.storage.clone();
+    let span = tracing::Span::current();
+
+    tokio::task::spawn_blocking(move || {
+        let _g = span.enter();
+        let mut connection = storage
+            .connection()
+            .context("Opening database connection")?;
+
+        let transaction = connection
+            .transaction()
+            .context("Creating database transaction")?;
+
+        let header = transaction
+            .block_header(block_id)
+            .context("Reading block from database")?
+            .ok_or(GetBlockError::BlockNotFound)?;
+
+        let l1_accepted = transaction.block_is_l1_accepted(header.number.into())?;
+        let block_status = if l1_accepted {
+            BlockStatus::AcceptedOnL1
+        } else {
+            BlockStatus::AcceptedOnL2
+        };
+
+        let transactions = get_block_transactions(&transaction, header.number)?;
+
+        Ok(types::Block::from_parts(header, block_status, transactions))
+    })
+    .await
+    .context("Database read panic or shutting down")?
+}
+
+/// This function assumes that the block ID is valid i.e. it won't check if the block hash or number exist.
+fn get_block_transactions(
+    db_tx: &pathfinder_storage::Transaction<'_>,
+    block_number: BlockNumber,
+) -> Result<Vec<TransactionWithHash>, GetBlockError> {
+    let txs = db_tx
+        .transaction_data_for_block(block_number.into())
+        .context("Reading transactions from database")?
+        .context("Transaction data missing for block")?
+        .into_iter()
+        .map(|(tx, _rx)| tx.into())
+        .collect();
+
+    Ok(txs)
+}
+
+mod types {
+    use crate::v02::types::reply::BlockStatus;
+    use crate::v04::types::TransactionWithHash;
+    use pathfinder_common::BlockHeader;
+    use serde::Serialize;
+    use serde_with::{serde_as, skip_serializing_none};
+
+    /// L2 Block as returned by the RPC API.
+    #[serde_as]
+    #[skip_serializing_none]
+    #[derive(Clone, Debug, Serialize, PartialEq, Eq)]
+    pub struct Block {
+        #[serde(flatten)]
+        pub header: crate::v05::types::BlockHeader,
+        pub status: BlockStatus,
+        pub transactions: Vec<TransactionWithHash>,
+    }
+
+    impl Block {
+        pub fn from_parts(
+            header: BlockHeader,
+            status: BlockStatus,
+            transactions: Vec<TransactionWithHash>,
+        ) -> Self {
+            Self {
+                header: header.into(),
+                status,
+                transactions,
+            }
+        }
+
+        /// Constructs [Block] from [sequencer's block representation](starknet_gateway_types::reply::Block)
+        pub fn from_sequencer(block: starknet_gateway_types::reply::MaybePendingBlock) -> Self {
+            Self {
+                status: block.status().into(),
+                transactions: block
+                    .transactions()
+                    .into_iter()
+                    .map(|t| t.clone().into())
```

Comment:
Nitpick, I find this nicer:
```
block.transactions().into_iter().map(Into::into)
```

Also the clone should be unnecessary? If it is, `.cloned()` is also nicer IMO

Reply:
Yeah, it's nicer. `.cloned()` is necessary though because `.transaction()` returns a slice.